ComponentKit/TransactionalDataSources/Common/Internal/CKTransactionalComponentDataSourceChangesetVerification.mm

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "CKTransactionalComponentDataSourceChangesetVerification.h"
+
+#import <ComponentKit/CKTransactionalComponentDataSourceChangesetInternal.h>
+#import <ComponentKit/CKTransactionalComponentDataSourceStateInternal.h>
+
+static NSMutableArray<NSNumber *> *sectionCountsForState(CKTransactionalComponentDataSourceState *state);
+
+static NSArray<NSIndexPath *> *sortedIndexPaths(NSArray<NSIndexPath *> *indexPaths);
+
+CKBadChangesetOperationType CKIsValidChangesetForState(CKTransactionalComponentDataSourceChangeset *changeset,
+                                                       CKTransactionalComponentDataSourceState *state)
+{
+  NSMutableArray<NSNumber *> *sectionCounts = sectionCountsForState(state);
+  __block BOOL invalidChangeFound = NO;
+  // Updated items
+  [changeset.updatedItems enumerateKeysAndObjectsUsingBlock:^(NSIndexPath * _Nonnull indexPath, id _Nonnull model, BOOL * _Nonnull stop) {
+    const NSInteger section = indexPath.section;
+    const NSInteger item = indexPath.item;
```

Comment:
For Index path? There should be no problem here. section/item work properly as you would expect.